insights/parsers/sys_module.py

```diff
@@ -0,0 +1,89 @@
+"""
+``/sys/module`` System Module Information
+=========================================
+
+A parser to parse the system module information.
+
+Parsers included in this module are:
+
+DMModUseBlkMq - file ``/sys/module/dm_mod/parameters/use_blk_mq``
+
+SCSIModUseBlkMq - file ``/sys/module/scsi_mod/parameters/use_blk_mq``
+---------------------------------------------------------------------
+
+"""
+
+
+from insights import parser, Parser
+from insights.parsers import ParseException, SkipException
+from insights.specs import Specs
+
+
+class XModUseBlkMq(Parser):
+    """
+    Parse for file `/sys/module/{dm_mod,scsi_mod}/parameters/use_blk_mq`.
+    File content shows if use_blk_mq parameter is on.
+
+    Sample Content::
+
+        Y
+
+    Raises:
+        SkipException: When nothing need to parse.
+
+    Attributes:
+        val(str): Raw data of the content.
+    """
+
+    def parse_content(self, content):
+        if not content or len(content) != 1:
+            raise SkipException()
+        self.val = content[0].strip()
+
+    @property
+    def is_on(self):
+        """
+        Returns:
+            (bool): True for on, False for off.
+
+        Raises:
+            ParseException: When tell is_on for unknown cases.
+        """
+        if self.val in ['Y', '1']:
+            return True
+        elif self.val in ['N', '0']:
+            return False
+        else:
+            raise ParseException("Unknown value: {0}".format(self.val))
```

Comment:
@JoySnow @xiangce my original comment was related to valid values and whether you want to be able to check for invalid values (not 'YN10') in a rule.  If that is important in a rule then this should probably be a `ValueError` or custom exception to indicate valid parsing with unexpected content.  An even better solution would be to use a negative or "[dirty](https://insights-core.readthedocs.io/en/latest/exception_model.html#case-3b-parsing-data-to-find-errors-dirty-parser)" parser which specifically looks for unexpected content and can be used by rules instead of exception handling.

If you are going to raise a `ParseException` then it should be raised in parse_content. See the [insights-core documentation section on exceptions](https://insights-core.readthedocs.io/en/latest/exception_model.html#case-3-unparsable-data-provides-useful-information) for more information.

Reply:
Hi @bfahr , I had updated the code to use  `ValueError.